src/irr/asset/CImageLoaderOpenEXR.cpp

```diff
@@ -0,0 +1,464 @@
+		{	
+			const size_t begginingOfFile = _file->getPos();
+
+			char magicNumberBuffer[sizeof(SContext::magicNumber)];
+			_file->read(magicNumberBuffer, sizeof(SContext::magicNumber));
+			_file->seek(begginingOfFile);
+
+			return isImfMagic(magicNumberBuffer);
+		}
+
+		template<typename rgbaFormat>
+		void readRgba(InputFile& file, std::array<Array2D<rgbaFormat>, 4>& pixelRgbaMapArray, int& width, int& height, E_FORMAT& format, const uint8_t& availableChannels)
+		{
+			Box2i dw = file.header().dataWindow();
+			width = dw.max.x - dw.min.x + 1;
+			height = dw.max.y - dw.min.y + 1;
+
+			constexpr char* rgbaSignatureAsText[] = {"R", "G", "B", "A"};
+			for (auto& pixelChannelBuffer : pixelRgbaMapArray)
+				pixelChannelBuffer.resizeErase(height, width);
+
+			FrameBuffer frameBuffer;
+			PixelType pixelType;
+
+			if (format == EF_R16G16B16A16_SFLOAT)
+				pixelType = PixelType::HALF;
+			else if (format == EF_R32G32B32A32_SFLOAT)
+				pixelType = PixelType::FLOAT;
+			else if (format == EF_R32G32B32A32_UINT)
+				pixelType = PixelType::UINT;
+
+			for (uint8_t rgbaChannelIndex = 0; rgbaChannelIndex < availableChannels; ++rgbaChannelIndex)
+				frameBuffer.insert
+				(
+					rgbaSignatureAsText[rgbaChannelIndex],                                                      // name
+					Slice(pixelType,                                                                            // type
+					(char*)(&(pixelRgbaMapArray[rgbaChannelIndex])[0][0] - dw.min.x - dw.min.y * width),        // base
+					sizeof((pixelRgbaMapArray[rgbaChannelIndex])[0][0]) * 1,                                    // xStride
+					sizeof((pixelRgbaMapArray[rgbaChannelIndex])[0][0]) * width,                                // yStride
+					1, 1,                                                                                       // x/y sampling
+					1                                                                                           // fillValue - default alpha if there is no alpha channel
+				));	
+
+			file.setFrameBuffer(frameBuffer);
+			file.readPixels(dw.min.y, dw.max.y);
+		}
+
+		bool specifyIrrlichtEndFormat(E_FORMAT& format, const InputFile& file, bool& doesItSupportAlphaChannel)
+		{
+			const IMF::Channel* RChannel = file.header().channels().findChannel("R");
+			const IMF::Channel* GChannel = file.header().channels().findChannel("G");
+			const IMF::Channel* BChannel = file.header().channels().findChannel("B");
+			const IMF::Channel* AChannel = file.header().channels().findChannel("A");
+
+			const IMF::Channel* XChannel = file.header().channels().findChannel("X");
+			const IMF::Channel* YChannel = file.header().channels().findChannel("Y");
+			const IMF::Channel* ZChannel = file.header().channels().findChannel("Z");
+
+			if (XChannel && YChannel && ZChannel)
+			{
+				os::Printer::log("LOAD EXR: the file consist of not supported CEI XYZ channels", file.fileName(), ELL_ERROR);
+				return false;
+			}
+			else if (!RChannel || !GChannel || !BChannel)
```

Comment:
we can support single R channel, RG channel and RGB channel

Reply:
but obvs no GB, G, B, RB etc.

Reply:
if no channel, what should I fill it with?

Reply:
there are R16 and R32 formats as well as RG16 and RG32, RGB is special cause it would be Not Power of Two (NPoT) so it doesnt exists and you use RGBA instead